matrix/matrix.py

```diff
@@ -0,0 +1,161 @@
+
+        Args:
+            operation (function): the function of the operator.
+            other (Matrix): the other matrix to operate on.
+
+        Returns:
+            Matrix: the result of the operation.
+
+        Note:
+            Boolean operations supports un-squared matrices but
+            mathematical operations don't.
+        """
+        if len(other.tuples) != len(self.tuples) and not any(
+                [operation == operator.eq, operation == operator.ne]):
+            raise ValueError("Matrices length don't match")
+
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(
+                    operation(self.tuples[i][j], other.tuples[i][j]))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(tuple(calculated_matrix))
+
+    def _by_scalar(self, scalar, operation):
+        """Calculate operation on matrix by scalar.
+
+        Args:
+            operation (function): the function of the operator.
+            scalar (number): the scalar to operate by.
+
+        Returns:
+            Matrix: the result of the operation.
+
+        Note:
+            Scalar must be a number. Operations with scalars are only * and /.
+        """
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(operation(self.tuples[i][j], scalar))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(tuple(calculated_matrix))
+
+    def __add__(self, other):
+        """Add two matrices."""
+        return self._matrix_by_matrix(other, operator.add)
+
+    def __sub__(self, other):
+        """Subtract two matrices."""
+        return self + (-1 * other)
+
+    def __mul__(self, other):
+        """Multiply two matrices or scalar by matrix."""
+        if isinstance(other, Matrix):
+            return self._matrix_by_matrix(other, operator.mul)
+
+        return self._by_scalar(other, operator.mul)
+
+    def __rmul__(self, other):
+        """Define to handle reverse multiplication (10 * matrix) as regular
+        multiplication."""
+        return self * other
+
+    def __truediv__(self, other):
+        """Divide matrix by scalar. Returns floats if
+        needed."""
+        return self * (1 / other)
+
+    def _check_bool_matrix(self, matrix, operation):
+        """Return true if all the items in the matrix are true else false."""
+        func = {operator.eq: all, operator.ne: any}.get(operation)
+        return func(
+            [item for line_data in matrix.tuples for item in line_data])
+
+    def __eq__(self, other):
+        """Compare two matrices. Returns true if they are equal else false."""
+        return self._check_bool_matrix(
```

Comment:
you're overcomplicating it, just compare the tuples
then delete methods like `_check_bool_matrix` and `_by_scalar` as they have exactly one usage